src/forwardBackwardIterations.jl

```diff
@@ -18,104 +22,148 @@ Parameters:
+                            solverProblems,
+                            forwardPassNumber::Int64,
+                            xi::Array{Float64, 3},
+                            returnCosts=true,
+                            init=false,
+                            display=false)
+
+    # TODO: verify that loops are in the same order
+    T = model.stageNumber
+    stocks = zeros(param.forwardPassNumber, T, model.dimStates)
+    # stocks[:, 1, :] = 90*rand(param.forwardPassNumber, model.dimStates)
+
+    costs = nothing
+    if returnCosts
+        costs = zeros(param.forwardPassNumber)
+    end
 
-for k = 1:forwardPassNumber #TODO can be parallelized + some can be dropped if too long
-    
-    for t=0:T-1 #TODO get T
-        stocks[k,t+1], opt_control = solveOneStepOneAlea(t,stocks[k,t],xi[k,t],
-                                    returnOptNextStage=true, 
-                                    returnOptControl=true,
-                                    returnSubgradient=false,
-                                    returnCost=false);
-        if returnCosts 
-            costs[k] += costFunction(t,stocks[k,t],opt_control,xi[k,t]); #TODO
+    for k = 1:param.forwardPassNumber
+
+        for t=1:T-1
+            state_t = extract_vector_from_3Dmatrix(stocks, t, k)
+            alea_t = extract_vector_from_3Dmatrix(xi, k, t)
+
+            status, nextstep = solve_one_step_one_alea(
+                                            model,
+                                            param,
+                                            solverProblems[t],
+                                            t,
+                                            state_t,
+                                            alea_t,
+                                            init)
+
+            stocks[k, t+1, :] = nextstep.next_state
+            opt_control = nextstep.optimal_control
+            if display
+                println(sizeof(opt_control))
+            end
+
+            if returnCosts
+                costs[k] += model.costFunctions(t, state_t, opt_control, alea_t)
+            end
         end
     end
+    return costs, stocks
 end
-return costs,stocks # adjust according to what is asked
-end
-
-
-
-
 
 
 
 """
-add to Vt a cut of the form Vt >= beta + <lambda,.>
+Add to Vt a cut of the form Vt >= beta + <lambda,.>
 
 Parameters:
-- Vt (bellmanFunction) 
+- Vt (bellmanFunction)
     Current lower approximation of the Bellman function at time t
-- beta (Float) 
+
+- beta (Float)
     affine part of the cut to add
+
 - lambda (Array{float,1})
     subgradient of the cut to add
-"""
-function addcut!(Vt::PolyhedralFunction, beta::Float, lambda::Array{float,1})
-    #TODO add >= beta + <lambda,.>,
-end
-
-
 
+    """
+function add_cut!(model, problem, Vt, beta::Float64, lambda::Array{Float64,1})
+    Vt.lambdas = vcat(Vt.lambdas, lambda)
+    Vt.betas = vcat(Vt.betas, beta)
+    Vt.numCuts += 1
+
+    alpha = getVar(problem, :alpha)
```

Comment:
Shouldn't we split this addCut! in two functions ? One for the Vt and for defining the model